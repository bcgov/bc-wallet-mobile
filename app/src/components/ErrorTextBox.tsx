import { useTheme } from '@hyperledger/aries-bifold-core'
import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

interface ErrorTextBoxProps {
  children: React.ReactElement | string
}

const iconSize = 30
const offset = 10

const ErrorTextBox: React.FC<ErrorTextBoxProps> = ({ children }) => {
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    container: {
      backgroundColor: ColorPallet.notification.error,
      borderRadius: 5,
      borderWidth: 1,
      borderColor: ColorPallet.notification.errorBorder,
      paddingHorizontal: 10,
      paddingVertical: 20,
    },
    row: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    textContainer: {
      ...TextTheme.normal,
      color: ColorPallet.notification.errorText,
      alignSelf: 'center',
      flexShrink: 1,
    },
    iconContainer: {
      marginRight: offset,
      alignSelf: 'flex-start',
    },
  })
  return (
    <View style={styles.container}>
      <View style={styles.row}>
        <View style={styles.iconContainer}>
          <Icon name={'error'} size={iconSize} color={ColorPallet.notification.errorIcon} />
        </View>
        {typeof children === 'string' ? <Text style={styles.textContainer}>{children}</Text> : <>{children}</>}
      </View>
    </View>
  )
}

export default ErrorTextBox
