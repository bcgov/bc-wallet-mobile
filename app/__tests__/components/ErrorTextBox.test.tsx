import { render } from '@testing-library/react-native'
import React from 'react'

import ErrorTextBox from '../../src/components/ErrorTextBox'

describe('ErrorTextBox Component', () => {
  test('renders correctly', () => {
    const tree = render(<ErrorTextBox>Lorem ipsum sit dolar</ErrorTextBox>)
    expect(tree).toMatchSnapshot()
  })
})
