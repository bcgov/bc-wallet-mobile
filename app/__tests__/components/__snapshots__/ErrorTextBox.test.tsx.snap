// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTextBox Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#313132",
      "borderColor": "#D8292F",
      "borderRadius": 5,
      "borderWidth": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignSelf": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#D8292F",
              "fontSize": 30,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î€€
      </Text>
    </View>
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontSize": 18,
          "fontWeight": "normal",
        }
      }
    >
      Lorem ipsum sit dolar
    </Text>
  </View>
</View>
`;
